Sources/Subs.php

```diff
@@ -4982,9 +4983,31 @@ function inet_dtop($bin)
 	if ($db_type == 'postgresql')
```

Comment:
For the sake of argument, let's pretend for a moment that it is somehow impossible to write data in the wrong type to a column in PostgreSQL. If we pretend that, then here is the logic of the function as it is now:

1. Are we using a PostgreSQL database?
  A. Yes.
    - Then assume $bin contains a plain IP address in a string, and return it.
  B. No.
    - Then assume $bin contains an IP address that has been packed into binary format, and blindly feed it to inet_ntop().

The problem here is that you are assuming that using MySQL somehow guarantees that $bin is an IP address in binary format. Using MySQL does guarantee that at all. As we've seen in the installer already, it is easy for data in the wrong type to be written to this column, and therefore there is no guarantee that $bin really will be what it should be. You must do something to check if $bin really is what it should be.

Of course, if we don't pretend that it is somehow impossible for PostgreSQL to contain bad data, then we need to check $bin in that case, too.

In short, just because the data _should be_ in a certain format does not mean that it _is_ in that format. You must check if it is or not.